internal/config/config.go

```diff
@@ -0,0 +1,30 @@
+package config
+
+import (
+	"flag"
+	"log"
+
+	"github.com/caarlos0/env/v6"
+)
+
+type Config struct {
+	ServerAddress string `env:"SERVER_ADDRESS" envDefault:"localhost:8080"`
+	BaseURL       string `env:"BASE_URL" envDefault:"http://localhost:8080"`
+}
+
+// ParseAndLoadConfig parses the configuration from environment variables and command-line flags
+func ParseAndLoadConfig() *Config {
+	cfg := &Config{}
+
+	// Parsing environment variables
+	if err := env.Parse(cfg); err != nil {
+		log.Fatalf("Failed to parse config from env: %v", err)
```

Comment:
[CHANGE REQUEST] Please return an error
https://github.com/uber-go/guide/blob/master/style.md#exit-in-main